plugins/rhacm/src/components/ClusterContext/ClusterContext.tsx

```diff
@@ -33,18 +35,31 @@ export const ClusterContextProvider = (props: any) => {
   );
   useDebounce(refresh, 10);
 
-  let errorBool = false;
-  /*
-  Error can either be undefined or Error(), if its undefined, no error happended
-  if cluster has some 'error' key, it means the API sent back an error response
-  */
-  if (error instanceof Error || 'error' in cluster) {
-    errorBool = true;
+  const isError = Boolean(error || 'error' in cluster)
+  if (isError) {
+    const falseStatus = {
+      status: {
+        available: false,
+        reason: '',
+      },
+    };
+    if (error) {
+      falseStatus.status.reason = error.message;
+    } else if ('error' in cluster) {
+      falseStatus.status.reason = (
+        cluster as unknown as ErrorResponseBody
+      ).error.message;
+    }
+    Object.assign(cluster, falseStatus);
   }
```

Comment:
Thanks for the detailed explanation!

The `error` property is now constructed either from an error response or an actual error and is part of the context. The `reason` property for the data is not being changed anywhere. If the data is still getting fetched or there is an error the `data` prop is `null`. If no error happens the prorp `error` is `null`.
The property `isError` is not actually used anywhere atm but I think it could be usefull in some cases so I left it there.